### ISSUE TITLE
Enforce code conventions

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#3708]
https://www.devbridge.com/articles/coding-best-practices/

Currently there aren't coding convention checks in place to enforce coding conventions for this project other than code reviews.
- .EditorConfig will do some of that voluntarily, but it can be overridden.
- If we end up switching to Github actions, we could kick off scripts to run.
- The URL above suggests using LINT to do the legwork here.

**Comment by SteveIreland** [comment]
Have you been spying on my code again?!?!?

**Comment by objecttothis** [comment]
Lol. No, it was a comment @jekkos made in a PR that @edwinbsmith started.

**Comment by SteveIreland** [comment]
I miss @daN4cat.  I keep expecting him to pop up at any moment to crack the whip on me to keep me in line.

**Comment by ekardian** [comment]
@daN4cat we r expecting

**Comment by objecttothis** [comment]
Moving to 3.5.0 so we can get CI4 out the door with 3.4.0

**Comment by jekkos** [comment]
The config is in the ci4 branch, no sure if sth else is needed to make this work. I tried to go through some of the GitHub actions config but could not find  any option to activate it. Lets check again once we merge to master.

**Comment by jekkos** [comment]
The linter seems to run now in github actions: https://github.com/opensourcepos/opensourcepos/actions/runs/9024670984/job/24798972386

**Comment by jekkos** [comment]
A next step now is to run the linter locally first to fix all indentation issues in our code.

**Comment by jekkos** [comment]
The linter can also be setup in PhpStorm directly: https://www.jetbrains.com/help/phpstorm/using-php-cs-fixer.html#installing-configuring-php-cs-fixer

**Comment by jekkos** [comment]
Once this is integrated then it should be documented. The CI check will then prevent code to not deviate from these rules.

**Comment by jekkos** [comment]
Curious to see what issues it will find: https://github.com/opensourcepos/opensourcepos/actions/runs/12714942531/job/35446147431

**Comment by objecttothis** [comment]
> https://github.com/dbfx/github-phplint
> @objecttothis I believe we need to add something to the CI pipeline that detects obvious code issues. In case of the requisitions, a method call was in place that did not exist. These type of issues can be easily catched with tools like this.

Yeah, PHPStorm complains about those things. I'd say as long as it's not too overzealous, a linter like that could be helpful. There are a handful of todo's in the code for ones I found.

**Comment by jekkos** [comment]
This way we can check PHP version compatbility. It has found plenty of issues with PHP7.4. Do we want to drop support for it?

**Comment by objecttothis** [comment]
Yes, we are dropping PHP 7.4 support.  It isn't compatible with 3.4.0

**Comment by objecttothis** [comment]
It's also been EOL for quite some time.

**Comment by objecttothis** [comment]
It's too bad it doesn't continue when it get's stuck on 7.4

**Comment by jekkos** [comment]
Related to this issue, I've created PR #4409 which converts the Travis CI build to GitHub Actions. The new workflow includes:

- Build and test with PHP 8.2
- PHPUnit tests with MariaDB container
- Docker image builds and pushes to Docker Hub
- Creation/update of the unstable release on master

This is part of the migration from Travis CI to GitHub Actions mentioned in the original issue. The existing linter workflow (main.yml) remains unchanged and will run alongside this new build workflow.
